tournamentcontrol/competition/tests/test_competition_admin.py

```diff
@@ -1560,3 +1560,33 @@ def test_edit_match_youtube_api_basic_guard_clause(self):
             match.refresh_from_db()
             self.assertFalse(match.live_stream)
             self.assertIsNone(match.external_identifier)
+    def test_undo_draw_with_dependent_matches(self):
+        """Test that undo_draw handles matches with eval_related dependencies correctly."""
+        # Create a stage with two rounds as described in the issue
+        stage = factories.StageFactory.create()
+        
+        # Round 1: Create two matches (1 vs 2)
+        match1 = factories.MatchFactory.create(stage=stage, round=1, label="Match 1")
+        
+        # Round 2: Create matches that depend on Round 1 results
+        # These matches will have eval_related fields pointing to match1
+        match2 = factories.MatchFactory.create(
+            stage=stage, 
+            round=2, 
+            label="Winner vs Loser",
+            home_team_eval_related=match1,  # Home team is winner of match1
+            away_team_eval_related=match1   # Away team is loser of match1
```

Comment:
This record should also have eval field of "W" on both sides. This is just the related match, without stating which side of the result.